core/c4diagram.go

```diff
@@ -0,0 +1,355 @@
+// package defines diagram as code using the graph input
+package core
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+	"unsafe"
+
+	"github.com/kislerdm/diagramastext/core/compression"
+)
+
+type optionsPlantUMLClient struct {
+	httpClient HttpClient
+}
+
+type clientPlantUML struct {
+	options optionsPlantUMLClient
+
+	baseURL string
+}
+
+func (c *clientPlantUML) Do(v DiagramGraph) (ResponseDiagram, error) {
+	code, err := diagramGraph2plantUMLCode(v)
```

Comment:
**suggestion** why not use the adapter pattern here to make this transformation more abstract? Otherwise, we will only be able to receive some `DiagramGraph` to use this client, making it hard to extend.

Reply:
Please make a suggestion. The context: `DiagramGraph` is the base internal contract for the module, it does not have to overwritten. It can be extended though through its attributes.